src/views/catalog/customize/components/CustomizeForms/CustomizeForm.tsx

```diff
@@ -23,7 +23,14 @@ export const CustomizeForm: React.FC<CustomizeFormProps> = ({ template }) => {
 
   const { onSubmit, loaded, error } = useCustomizeFormSubmit({ template });
   const [requiredFields, optionalFields] = buildFields(template);
-  const nameField = getVirtualMachineNameField(template, t);
+
+  const [vmName, setVmName] = React.useState('');
+  React.useMemo(() => {
+    getvmName(template).then((name) => {
+      setVmName(name);
+    });
+  }, [template]);
```

Comment:
the use state here can be removed, you can use instead:
`const vmName = useMemo(() => getVMName(template) , [template])`
- we recently made a change that allows the following import statement: `import { useMemo } from 'react'`
- I'd think it is better to change getvmName to getVMName since VM is an acronym and it's a convention to write them as upper case.

Reply:
Thanks for the suggestions, they all make sense to me :)

Reply:
Note that my _getvmName_ returns a promise, so I am not sure if we can really get rid of using state to get the string for the VM name...